ckan/config/middleware/flask_app.py

```diff
@@ -132,6 +133,13 @@ def make_flask_stack(conf: Union[Config, CKANConfig]) -> CKANApp:
     app.template_folder = os.path.join(root, 'templates')
     app.app_ctx_globals_class = CKAN_AppCtxGlobals
 
+    application_root = config["ckan.root_path"] or "/"
+    if not application_root.endswith("/"):
+        application_root += "/"
+    application_root = application_root.replace("/{{LANG}}", "")
+    config["APPLICATION_ROOT"] = application_root
```

Comment:
this won't work if the `/{{LANG}}` part isn't at the end of `root_path`

Reply:
I missed the point, what is wrong?

* `/data/{{lang}}` -> append / `/data/{{lang}}/` -> drop lang with preceding / `/data`
* `/data/{{lang}}/other` -> append / `/data/{{lang}}/other/` -> drop lang with preceding / `/data/other/`